--- origsrc/gnome-mplayer-1.0.3/configure.in	2011-04-21 09:55:25.000000000 -0500
+++ src/gnome-mplayer-1.0.3/configure.in	2011-05-17 23:06:21.588668000 -0500
@@ -16,7 +16,7 @@ AC_HEADER_STDC
 AC_PROG_CPP
 AC_PROG_CXX
 AM_PROG_CC_C_O
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 dnl Set gettext package name
 GETTEXT_PACKAGE=gnome-mplayer
--- origsrc/gnome-mplayer-1.0.3/src/Makefile.am	2011-04-13 15:24:14.000000000 -0500
+++ src/gnome-mplayer-1.0.3/src/Makefile.am	2011-05-17 23:06:14.059237400 -0500
@@ -84,24 +84,23 @@ gnome_mplayer_LDADD = \
 if HAVE_NAUTILUS
 
 nautilusdir = $(NAUTILUSDIR)
-nautilus_PROGRAMS = libgnome-mplayer-properties-page.so
+nautilus_LTLIBRARIES = libgnome-mplayer-properties-page.la
 
-libgnome_mplayer_properties_page_so_SOURCES = \
+libgnome_mplayer_properties_page_la_SOURCES = \
     ../config.h \
     mime_types.h \
     nautilus_property_page.c
 
-libgnome_mplayer_properties_page_so_CFLAGS = -fPIC \
+libgnome_mplayer_properties_page_la_CFLAGS = -fPIC \
     $(NAUTILUS_CFLAGS) \
-	$(GTK_CFLAGS) \
     $(GIO_CFLAGS) \
     $(AM_CFLAGS)
     
-libgnome_mplayer_properties_page_so_LDFLAGS = -shared
+libgnome_mplayer_properties_page_la_LDFLAGS = \
+	-module -avoid-version -no-undefined
 
-libgnome_mplayer_properties_page_so_LDADD = \
+libgnome_mplayer_properties_page_la_LIBADD = \
 	$(NAUTILUS_LIBS) \
-	$(GTK_LIBS) \
 	$(GIO_LIBS)
 
 endif #HAVE_NAUTILUS
